Alpaca-Finance-Agents/HighShareLP/src/agent.spec.ts

```diff
@@ -0,0 +1,229 @@
+import { HandleBlock } from "forta-agent";
+import { provideHandleBlock } from "./agent";
+import {
+  createAddress,
+  TestBlockEvent,
+  encodeParameter,
+  encodeParameters,
+} from "forta-agent-tools";
+import { when, resetAllWhenMocks } from "jest-when";
+import {
+  isCallToTotalSupply,
+  isCallToLpToken,
+  isCallToPid,
+  isCallToBscPool,
+  isCallToWexMaster,
+  isCallToMasterChef,
+  isCallToUserInfo,
+} from "./mock.utils";
+import { constants } from "ethers";
+import { createFinding } from "./utils";
+
+const workers = [
+  createAddress("0x1"),
+  createAddress("0x2"),
+  createAddress("0x3"),
+  createAddress("0x047"),
+];
+
+const lpTokens = [
+  createAddress("0x5"),
+  createAddress("0x6"),
+  createAddress("0x7"),
+];
+
+const masterChef = createAddress("0x8");
+const bscPool = createAddress("0x9");
+const wexPool = createAddress("0x10");
+
+const addressesRegistry = {
+  vaults: [
+    {
+      workers: [
+        {
+          address: workers[0],
+          lpToken: lpTokens[0],
+        },
+        {
+          address: workers[1],
+          lpToken: lpTokens[1],
+        },
+      ],
+    },
+    {
+      workers: [
+        {
+          address: workers[2],
+          lpToken: lpTokens[2],
+        },
+        {
+          address: workers[3],
+          lpToken: constants.AddressZero, // Disabled worker
+        },
+      ],
+    },
+  ],
+};
+
+const mockAddressFetcher = () => addressesRegistry;
+
+const mockCall = jest.fn();
+const mockEthers = {
+  call: mockCall,
+  _isProvider: true, // Necessary for being an ethers provider
+} as any;
+
+const setTotalSuppliesValues = (
+  token1: string,
+  token2: string,
+  token3: string
+) => {
+  when(mockCall)
+    .calledWith(isCallToTotalSupply(lpTokens[0]), expect.anything())
+    .mockReturnValue(encodeParameter("uint256", token1));
+  when(mockCall)
+    .calledWith(isCallToTotalSupply(lpTokens[1]), expect.anything())
+    .mockReturnValue(encodeParameter("uint256", token2));
+  when(mockCall)
+    .calledWith(isCallToTotalSupply(lpTokens[2]), expect.anything())
+    .mockReturnValue(encodeParameter("uint256", token3));
+};
+
+describe("Worker With Too Much Shares Test Suite", () => {
```

Comment:
```suggestion
describe("Worker With More Than 51% of Shares Test Suite", () => {
```
Suggestion, if being more specific is appropriate.

Reply:
Not adding this because what is too much really depends on the argument used, and in test that is more probably to change.